3.3 release work

osfameron opened the pull request:
NB: I've not been able to install .NET 3.3 as yet, and may need to wipe and reinstall .NET entirely 😡 

So I've not reviewed the Transactions or Getting Started code in detail, but have aligned the Cloud.csx example with StartUsing.cs as we've done elsewhere.

osfameron committed:
3.3-release-prep wip

osfameron committed:
rename Cloud.csx 3.3-release-prep

osfameron committed:
3.3-release-prep update start using

Also add Platform Help to nav

osfameron committed:
Change @server version to 7.0 for now

As per https://github.com/couchbase/docs-sdk-java/pull/326

    git grep -l '{version-server}@server' | xargs sd '\{version-server\}@server' '7.0@server'

RichardSmedley commented on modules/hello-world/pages/start-using-sdk.adoc:
```diff
@@ -25,12 +25,14 @@ We will go through the code sample step by step, but for those in a hurry to see
 +
 --
-If you are connecting to https://docs.couchbase.com/cloud/index.html[Couchbase Capella], be sure to get the correct endpoint as well as user, password, and `couchbasecloudbucket` -- and see the <<cloud-connections, Cloud section>>, below.
+If you are connecting to https://docs.couchbase.com/cloud/index.html[Couchbase Capella], be sure to get the correct endpoint as well as user, password -- and see the <<cloud-connections, Cloud section>>, below.
```
We know longer have the Cloud section - we just need to give the C# equivalent of 
timeout_opts = ClusterTimeoutOptions(kv_timeout=timedelta(seconds=10))

[which, I realise, we also have to add to the Scala and Java docs]

RichardSmedley replied:
... but you can also link to the #troubleshooting section at the end, which contains what the cloud-connections section originally held.

osfameron replied:
so on balance all OK, or needs to be done? 😕

osfameron replied:
(Done in "Move to Cloud.cs" commit)

maria-robobug commented on modules/devguide/examples/dotnet/Cloud.csx:
```diff
@@ -0,0 +1,60 @@
+#r "nuget: CouchbaseNetClient, 3.2.8-pre"
+// #r "nuget: System.Text.Json, 6.0.2"
+// #r "nuget: Microsoft.Extensions.Logging.Abstractions, 6.0.1"
+// #r "nuget: CouchbaseNetClient, 3.3.0"
```
Since 3.3.0 is already available: https://www.nuget.org/packages/CouchbaseNetClient/3.3.0
Should be able to pull that in now?

Ah my bad, clicked on release note link and thought it was published, but nope (ignore me).

osfameron replied:
>  The specified version 3.3.0 was not found. You have been taken to version 3.2.9.
yup, schroedinger's release artefact 😂

RichardSmedley commented on modules/hello-world/pages/start-using-sdk.adoc:
```diff
@@ -25,12 +25,14 @@ We will go through the code sample step by step, but for those in a hurry to see
+include::devguide:example$dotnet/Cloud.csx[tags=**]
 ----
+
+The Couchbase Capella free trial version comes with the Travel Sample Bucket, and its Query indexes, loaded and ready.
```
This is after the tabs in other SDKs - it helps reinforce the Capella is easier message ;)

osfameron replied:
ah, I moved it inside (here and Java I think?) as it didn't make much sense to me after the tabs for the local server version

maria-robobug reviewed (approved):
👍

osfameron committed:
Move to Cloud.cs

`dotnet-script` isn't happy with SDK 3.3.0 versions of
Microsoft.Extensions.Logging.Abstractions 6.0.0 vs 6.0.1
but running with your own .csproj has no problem.

In own subdirectory, to get the primary benefit of script,
that is, not having to compile *all the examples* just to
run one example.

osfameron committed:
Dockerfile to run SDK examples

USAGE:

  docker-compose run cb-dotnet-sdk bash

osfameron committed:
Cloud start-using wording tweaks

jeffrymorris commented:
@osfameron ping me if your still having trouble installing 3.3.x
